src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs

```diff
@@ -166,14 +167,20 @@ public async Task ScriptStepAsync(IExecutionContext context, ContainerInfo conta
         }
 
         private async Task<HookResponse> ExecuteHookScript(IExecutionContext context, HookInput input, ActionRunStage stage)
+        {
+            var prependPath = string.Join(Path.PathSeparator.ToString(), context.Global.PrependPath.Reverse<string>());
+            return await ExecuteHookScript(context, input, stage, prependPath);
+        }
+
+        private async Task<HookResponse> ExecuteHookScript(IExecutionContext context, HookInput input, ActionRunStage stage, string prependPath)
```

Comment:
Let's actually make `string prependPath` a mandatory parameter, WhichUtil will always (potentially) need it.

We could have a private `GetPrependPath` method that does `var prependPath = string.Join(Path.PathSeparator.ToString(), context.Global.PrependPath.Reverse<string>());` in three of the hooks except in `ScriptStepAsync` which already has prependPath.